test/Microsoft.AspNetCore.Blazor.Build.Test/ComponentRenderingRazorIntegrationTest.cs

```diff
@@ -554,5 +554,57 @@ public class MyComponent : BlazorComponent
                 frame => AssertFrame.Text(frame, "hi", 14),
                 frame => AssertFrame.Markup(frame, "\n      <div></div>\n  ", 15));
         }
+
+        [Fact]
+        public void RazorTemplate_CanBeUsedFromComponent()
+        {
+            // Arrange
+            AdditionalSyntaxTrees.Add(Parse(@"
+using Microsoft.AspNetCore.Blazor;
+using Microsoft.AspNetCore.Blazor.Components;
+using Microsoft.AspNetCore.Blazor.RenderTree;
+
+namespace Test
+{
+    public class Repeater : BlazorComponent
+    {
+        [Parameter] int Count { get; set; }
+        [Parameter] RenderFragment<string> Template { get; set; }
+        [Parameter] string Value { get; set; }
+
+        protected override void BuildRenderTree(RenderTreeBuilder builder)
+        {
+            base.BuildRenderTree(builder);
+            for (var i = 0; i < Count; i++)
+            {
+                builder.AddContent(i, Template, Value);
+            }
+        }
+    }
+}
+"));
+
+            var component = CompileToComponent(@"
+@addTagHelper ""*, TestAssembly""
+<Repeater Count=3 Value=""Hello, World!"" Template=""@(@<div>@item.ToLower()</div>)"" />
+");
+
+            // Act
+            var frames = GetRenderTree(component);
+
+            // Assert
+            Assert.Collection(
+                frames,
+                frame => AssertFrame.Component(frame, "Test.Repeater", 4, 0),
+                frame => AssertFrame.Attribute(frame, "Count", typeof(int), 1),
+                frame => AssertFrame.Attribute(frame, "Value", typeof(string), 2),
+                frame => AssertFrame.Attribute(frame, "Template", typeof(RenderFragment<string>), 3),
+                frame => AssertFrame.Element(frame, "div", 2, 4),
+                frame => AssertFrame.Text(frame, "hello, world!", 5),
+                frame => AssertFrame.Element(frame, "div", 2, 4),
+                frame => AssertFrame.Text(frame, "hello, world!", 5),
+                frame => AssertFrame.Element(frame, "div", 2, 4),
+                frame => AssertFrame.Text(frame, "hello, world!", 5));
```

Comment:
@SteveSandersonMS - this doesn't seem right to me but I'm not sure what I would do differently. I expected to see these elements and text inside a frame for `<Repeater>`

Reply:
They are inside a frame for `<Repeater>` - that's the first item in the collection. Its children are the output from the `Repeater` component.

What I don't understand is why the children aren't grouped in `Region` frames. Your new `AddContent` overload does emit region delimiters, so why don't they appear in the output here?

Reply:
OK I'll dig into this and try to understand more.

Reply:
So from what I can understand the `GetRenderTree` method used by these tests will never include regions, since it's the list of edits that would get applied to an 'empty' tree. I can confirm that the regiions are part of the reference frames, but aren't part of the 'render batch'. This is as far as I got because the terms are all a bit abstract to me. 

We don't have any tests written in this style that include/verify regions. All of the tests that do verify regions get them directly from `RenderTreeBuilder.GetFrames()` with the exception of PassesThroughRegionsInsidePrependedElements

Let me know if you think this needs more investigation or follow up.